spec/requests/api/v1/users/update_spec.rb

```diff
@@ -0,0 +1,58 @@
+require 'rails_helper'
+
+describe 'PUT api/v1/users/:id', type: :request do
+  let(:user)              { create(:user) }
+  let(:new_email)         { 'test2@test.com' }
+  let(:new_name)          { 'Test2' }
+  let(:new_gender)        { 'female' }
+
+  context 'with valid params' do
+    let(:params) do
+      {
+        user: { email: new_email, name: new_name, gender: new_gender }
+      }
+    end
+
+    it 'returns a succesful response' do
+      put api_v1_user_path(user), params: params, headers: auth_headers, as: :json
+      expect(response).to have_http_status(:success)
+    end
+
+    it 'updates the user' do
+      put api_v1_user_path(user), params: params, headers: auth_headers, as: :json
+      user.reload
+      expect(user.email).to eq(new_email)
+      expect(user.name).to eq(new_name)
+      expect(user.gender).to eq(new_gender)
+    end
+
+    it 'returns the user' do
+      put api_v1_user_path(user), params: params, headers: auth_headers, as: :json
+      expect(json[:user]).to include_json(
+        id: user.id,
+        email: new_email,
+        name: new_name,
+        gender: new_gender
+      )
+    end
+  end
+
+  context 'with invalid params' do
+    let(:params) do
+      {
+        user: { email: 'invalid_email' }
+      }
+    end
+
+    it 'does not return success' do
+      put api_v1_user_path(user), params: params, headers: auth_headers, as: :json
+      expect(response).not_to have_http_status(:success)
```

Comment:
Expect `bad_request` of whatever the error is, but prefer to be accurate on the expectation